barretenberg/ts/src/barretenberg/backend.ts

```diff
@@ -433,3 +456,39 @@ function base64Decode(input: string): Uint8Array {
     throw new Error('No implementation found for base64 decoding.');
   }
 }
+
+/**
+ * Helper function to wrap promises with abort signal support
+ * @param signal AbortSignal to abort the operation
+ * @param fn Function that returns a promise
+ */
+function withAbort<T>(signal: AbortSignal | undefined, fn: () => Promise<T>): Promise<T> {
+  if (signal?.aborted) {
+    return Promise.reject(new DOMException('Proving aborted', 'AbortError'));
```

Comment:
I don't think there is a requirement to throw a Dom exception here! A normal exception is perferred, Overall pattern here is good

Reply:
Yep, that is preferred as this will run in node js too!